mipengine/controller/api/DTOs/AlgorithmSpecificationsDTOs.py

```diff
@@ -0,0 +1,150 @@
+from dataclasses import dataclass
+from typing import List, Optional, Dict
+
+from dataclasses_json import dataclass_json
+
+from mipengine.controller.common.algorithms_specifications import CROSSVALIDATION_ALGORITHM_NAME, \
+    GenericParameterSpecification, AlgorithmSpecifications, AlgorithmsSpecifications
+from mipengine.controller.common.utils import Singleton
+
+INPUTDATA_PATHOLOGY_PARAMETER_NAME = "pathology"
+INPUTDATA_DATASET_PARAMETER_NAME = "dataset"
+INPUTDATA_FILTERS_PARAMETER_NAME = "filter"
+INPUTDATA_X_PARAMETER_NAME = "x"
+INPUTDATA_Y_PARAMETER_NAME = "y"
+
+
+@dataclass_json
+@dataclass
+class InputDataSpecificationDTO:
+    """
+    InputDataParameterDTO is different from the InputDataParameter
+    on the stattypes field.
+    It is optional on the DTOs, due to the dataset and pathology parameters.
+    """
+    label: str
+    desc: str
+    types: List[str]
+    notblank: bool
+    multiple: bool
+    stattypes: Optional[List[str]] = None
+    enumslen: Optional[int] = None
+
+
+def get_pathology_parameter():
+    return InputDataSpecificationDTO(
+        "Pathology of the data.",
+        "The pathology that the algorithm will run on.",
+        ["text"],
+        True,
+        False,
+        None,
+        None,
+    )
+
+
+def get_dataset_parameter():
+    return InputDataSpecificationDTO(
+        "Set of data to use.",
+        "The set of data to run the algorithm on.",
+        ["text"],
+        True,
+        True,
+        None,
+        None,
+    )
+
+
+def get_filter_parameter():
+    return InputDataSpecificationDTO(
+        "Filter on the data.",
+        "Features used in my algorithm.",
+        ["jsonObject"],
+        False,
+        False,
+        None,
+        None,
+    )
+
+
+@dataclass_json
+@dataclass
+class GenericParameterSpecificationDTO(GenericParameterSpecification):
+    """
+    GenericParameterDTO is identical to the GenericParameterSpecification
+    but exists for consistency and future use if needed.
+    """
+    pass
+
+
+@dataclass_json
+@dataclass
+class CrossValidationSpecificationsDTO:
+    """
+    CrossValidationDTO is a nested object, that contains
+    all the information need to run crossvalidation on an algorithm.
+    """
+    desc: str
+    label: str
+    parameters: Dict[str, GenericParameterSpecification]
+
+
+@dataclass_json
+@dataclass
+class AlgorithmSpecificationDTO:
+    """
+    AlgorithmDTO is used to provide the UI the requirements
+    of each algorithm.
+    System variables are added and unnecessary fields are removed.
+    """
+    name: str
+    desc: str
+    label: str
+    inputdata: Dict[str, InputDataSpecificationDTO]
+    parameters: Optional[Dict[str, GenericParameterSpecification]] = None
+    crossvalidation: Optional[CrossValidationSpecificationsDTO] = None
```

Comment:
I'm not sure crossvalidation belongs here. Cross-validation is not part of one algorithm. It can be part of one experiment. One experiment (to be defined) is a larger structure having one or more Algorithms in pipeline or other structure, some data specs, some evaluation strategies (e.g. cross-validation), some visualizations, etc...

Reply:
For the February demo/release/mvp we are only going to have algorithms with or without cross validation, no pipeline.

In the future this could be renamed to ExperimentSpecifications or something different and hold more information.

We can make changes accordingly when we have decided about the whole experiment/pipeline logic.